subsys/mgmt/mcumgr/smp_pkasm.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_MGMT_SMP_PKASM_H_
+#define ZEPHYR_INCLUDE_MGMT_SMP_PKASM_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct zephyr_smp_transport;
```

Comment:
Please add header dependencies here (e.g header with _zephyr_smp_transport_ definition)

Reply:
I am not sure about this. The re-assembly code is internal SMP thing that is going to be included only to files that already import the mentioned headers, so this will only duplicate the code. The header has been provided as separate unit to make it easier to figure out which set of functions is re-assembly "API".